### PR TITLE
[AKS]: Remove support for `az aks mesh` egress gateway commands

### PR DESCRIPTION
**Pull request by deveshdama**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
`az aks mesh enable-egress-gateway -n cluster-name -g rg-name`
`az aks mesh disable-egress-gateway -n cluster-name -g rg-name`

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -11,6 +11,10 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 +++++++
+
+5.0.0b5
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]:
```suggestion
6.0.0b1
```

File: src/aks-preview/setup.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 
 from setuptools import setup, find_packages
 
-VERSION = "5.0.0b4"
+VERSION = "5.0.0b5"
```
Comment (FumingZhang) [inline src/aks-preview/setup.py:12]:
```suggestion
VERSION = "6.0.0b1"
```

**Review by FumingZhang — COMMENTED** [review]
lgtm, but should update the major version as this is a breaking change.

File: src/aks-preview/azext_aks_preview/commands.py [hunk 3]
```diff
@@ -396,23 +396,12 @@ def load_command_table(self, _):
             "aks_mesh_enable_ingress_gateway",
             supports_no_wait=True,
         )
-        g.custom_command(
-            "enable-egress-gateway",
-            "aks_mesh_enable_egress_gateway",
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/commands.py:401]: Oh, I think you should also delete the command implementations, like function `aks_mesh_disable_egress_gateway` in `custom.py`

Comment (nshankar13) [reply src/aks-preview/azext_aks_preview/commands.py:401]: Would like to keep this if possible as we will likely reintroduce egress gateway in the near future.

**Review by nshankar13 — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/commands.py (hunk 3 above)
Comment (deveshdama) [reply src/aks-preview/azext_aks_preview/commands.py:401]: I agree with Niranjan here, we would like to keep it as we see that egress gateway functionality will be reintroduced. This PR is to just remove user/test facing code components. @FumingZhang let us know your thoughts?

**Review by SanyaKochhar — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/commands.py (hunk 3 above)
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/commands.py:401]: ok, get it

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**